packages/dapp/src/components/parameterizer/proposeReparameterization.tsx

```diff
@@ -64,6 +64,15 @@ export class ProposeReparameterization extends React.Component<
     );
   }
 
+  public componentWillReceiveProps(nextProps: ProposeReparameterizationProps): void {
```

Comment:
I'd prefer if you changed this to `getDerivedStateFromProps` now.  The only difference is that you return the object instead of passing it to setState

Reply:
Sure. I wasn't sure if the method was available in our current version of React, but I can make this change.

Reply:
I guess your right it was added in 16.3.  We should upgrade to that. Its been out for couple months now

Reply:
@walfly: is upgrading a blocker for this PR? 

I'd prefer to do an upgrade as a separate PR, since there might be a bunch of changes to get in line with the new React methods.

Reply:
its a minor so it will just add warnings, but go ahead and do it separately.